test/dynamo/test_ctx_manager.py

```diff
@@ -1,13 +1,40 @@
 # Owner(s): ["module: dynamo"]
+import contextlib
+import sys
+import traceback
 import unittest
+from contextlib import contextmanager
 
 import torch
 import torch._dynamo.test_case
 import torch._dynamo.testing
 from torch._dynamo.testing import EagerAndRecordGraphs, normalize_gm, same
+from torch._dynamo.utils import counters
 from torch.nn import functional as F
 from torch.testing._internal.common_cuda import PLATFORM_SUPPORTS_FLASH_ATTENTION
-from torch.testing._internal.common_utils import TEST_WITH_ROCM
+from torch.testing._internal.common_utils import (
+    instantiate_parametrized_tests,
+    parametrize,
+    TEST_WITH_ROCM,
+)
+
```

Comment:
I checked out the PR locally. I am getting an incorrect result from the following script:
```
import torch
import contextlib

state = 0

@contextlib.contextmanager
def change_state(val): # E: Function is missing a type annot
    global state
    old_state = state
    try:
        state = val
        yield
    finally:
        state = old_state

@torch.compile(backend="eager", fullgraph=True)
def f(x): # E: Function is missing a type annotation  [no-un
    with change_state(3):
        return x * state

x = torch.tensor(1.)
print(f(x))
```

Reply:
This is the same error in the test `test_change_parent_global_0`